tests/integration/cattletest/core/test_projects.py

```diff
@@ -0,0 +1,418 @@
+    assert e.value.error.status == 404
+
+
+def test_get_members(project_clients, members):
+    project = _create_project_with_members(project_clients['Owner'], members)
+    members = project.projectMembers()
+    _get_members(project_clients['Owner'], project.id, members)
+    _get_members(project_clients['Member'], project.id, members)
+    _get_members(project_clients['admin'], project.id, members)
+    with pytest.raises(ApiError) as e:
+        _get_members(project_clients['Stranger'], project.id, members)
+    assert e.value.error.status == 404
+
+
+def test_list_all_projects():
+    projects = _admin_client().list_project()
+    projectAccounts = _admin_client().list_account(kind='project', limit=4000)
+    ids = []
+    ids_2 = []
+    for project in projects:
+        ids.append(project.id)
+    for project in projectAccounts:
+        ids_2.append(project.id)
+    assert len(list(set(ids) - set(ids_2))) == 0
+
+
+def _create_resources(super_client, user_account, client):
+    create_sim_context(super_client, 'simagent' + random_str(),
+                       account=user_account, public=True)
+    for x in range(0, 4):
+        uuid = "sim:{}".format(random_num())
+        client.wait_success(client.create_container(imageUuid=uuid))
+    registry = client.create_registry(serverAddress='quay.io',
+                                      name='Quay')
+    registry = client.wait_success(registry)
+    reg_cred = client.create_registry_credential(
+        registryId=registry.id,
+        email='test@rancher.com',
+        publicValue='rancher',
+        secretValue='rancher')
+    client.wait_success(reg_cred)
+
+
+def check_state(client, project_id, states, excludes):
+    for type in client.schema.types:
+        try:
+            for resource in client.list(type, accountId=project_id):
+                if resource.type not in excludes:
+                    assert resource.state in states
+                    assert resource.removed is not None
+        except AttributeError:
+            pass
+
+
+def test_delete_project(project_clients, members, super_client):
+    project = _create_project_with_members(project_clients['Owner'],
+                                           members=members)
+    proj_id = project.id
+    client = client_for_project(project)
+    _create_resources(super_client, project, client)
+    assert len(client.list_projectMember()) == 2
+    project = super_client.wait_success(project.deactivate())
+    project = super_client.wait_success(project.remove())
+    check_state(project_clients['Owner'], proj_id,
+                ['removed'], ['account', 'project', 'host'])
+    super_client.wait_success(project.purge())
+    project = project_clients['Owner'].by_id('project', id=proj_id)
+    assert project.state == 'purged'
+    check_state(project_clients['Owner'], proj_id,
+                ['purged', 'removed'], ['account', 'project'])
+    project_members = project_clients['admin']\
+        .list('projectMember')
```

Comment:
James, should we also check here that all the resources (containers) created by _create_resources, were deleted along with the project delete?